File: src/SoftielRemote.App.Flutter/lib/models/connection_status.dart
/// Connection status enum
enum ConnectionStatus {
  disconnected,
  connecting,
  connected,
  error,
}

extension ConnectionStatusExtension on ConnectionStatus {
  String get displayName {
    switch (this) {
      case ConnectionStatus.disconnected:
        return 'Bağlantı Yok';
      case ConnectionStatus.connecting:
        return 'Bağlanıyor...';
      case ConnectionStatus.connected:
        return 'Bağlı';
      case ConnectionStatus.error:
        return 'Hata';
    }
  }
}

File: legacy/src/SoftielRemote.Core/Enums/ConnectionStatus.cs
namespace SoftielRemote.Core.Enums;

/// <summary>
/// Bağlantı durumunu temsil eder.
/// </summary>
public enum ConnectionStatus
{
    /// <summary>
    /// Bağlantı kurulmamış.
    /// </summary>
    Disconnected = 0,

    /// <summary>
    /// Bağlantı isteği gönderildi, bekleniyor.
    /// </summary>
    Pending = 1,

    /// <summary>
    /// Bağlantı kabul edildi, kuruluyor.
    /// </summary>
    Connecting = 2,

    /// <summary>
    /// Bağlantı aktif ve çalışıyor.
    /// </summary>
    Connected = 3,

    /// <summary>
    /// Bağlantı reddedildi.
    /// </summary>
    Rejected = 4,

    /// <summary>
    /// Bağlantı hatası oluştu.
    /// </summary>
    Error = 5
}

